modules/self-test/test_declare.lua

```diff
@@ -36,7 +36,14 @@
 			error('Duplicate test suite "'.. suiteName .. '"', 2)
 		end
 
-		local suite = {}
+		local _suite = {}
+		-- Setup a metatable for the test suites to use, this will catch duplicate tests
+		local suite = setmetatable({}, {
+			__index = _suite,
+			__newindex = function (table, key, value) if m.detectDuplicateTests and _suite[key] ~= nil then error('Duplicate test "'.. key .. '"', 2) end _suite[key] = value end,
```

Comment:
can we use normal multi-line code... I know it's just a styling thing, but it makes it a bit more readable.
also `p.error`.

Reply:
Ah, my bad, I wasn't supposed to leave it like that.

Not sure why I used `error` over `p.error`, weird.

Reply:
Oh, the top of this function uses `error`, that's why I used it. `p.error` doesn't provide the file location either, whereas `error` does, which is likely why `error` is used instead here.

Reply:
Example using `error`:
```
C:\github\premake-core>bin\release\premake5.exe test
Running action 'test'...
Error: ** Error: Duplicate test "referencesAreRelative_onDifferentProjectLocation"
```
Example using `p.error`:
```
C:\github\premake-core>bin\release\premake5.exe test
Running action 'test'...
Error: ...-core/modules/vstudio/tests/vc2010/test_project_refs.lua:83: Duplicate test "referencesAreRelative_onDifferentProjectLocation"
```